pkg/postage/stamper.go

```diff
@@ -21,7 +21,9 @@ var (
 
 // Stamper can issue stamps from the given address of chunk.
 type Stamper interface {
-	Stamp(swarm.Address) (*Stamp, error)
+	// addr is the request address of the chunk and idAddr is the identity address of the chunk.
+	Stamp(addr, idAddr swarm.Address) (*Stamp, error)
```

Comment:
Let's have a new Stamp function, maybe called StampSOC that's specific to this case.

Reply:
So we can have the original function definition untouched

Reply:
later it may change to have only one param but with idAddress.
I do not see the reason to complicate this further there should be only one stamp funtion